Sources/Display.php

```diff
@@ -1134,6 +1134,30 @@ function Display()
 				array('id_member', 'id_board')
 			);
 		}
+
+		// Mark any alerts about this topic or the posts on this page as read.
+		$smcFunc['db_query']('', '
```

Comment:
got we a chance to know if this update is needed or not,
without doing a select?

Reply:
There's no need to do a SELECT. If the user is reading a topic, then the alerts about that topic should be marked as read.

The WHERE clause will mark alerts as read that are (1) about the topic as a whole (e.g. locked, stickied, etc.) and (2) about messages on the specific page that the user if viewing (e.g. likes, mentions, etc.). In both cases, if the page is being shown to the user, then by definition the content of those alerts has been read and the alerts should be marked as read.

This is the same logic used by social media sites, GitHub, etc., who marks alerts about content on the site as read when the user views the content, even if the user didn't look at the alert page itself to get to the content.

Make sense?

Reply:
github and co. used here an async query,
since we doesn't got something like that in php,
i asked if we know if the update query is needed to run.

Reply:
Oh, I see what you mean. 

No, I just run the query directly during page load because it is more efficient to use the information we already have than it would be to use a separate AJAX request. 

Using AJAX for this task would require the server to redo all the work to get this information again, and then we'd still need to run a very similar query to mark the alerts as read anyway. 

By running this query now, we skip all that extra work and make the most efficient use of the work we have already done.

Reply:
I suppose we could add an `if (!empty($user_info['alerts']))` before running the query, since there's no point in running it if the user doesn't have any unread alerts.